### DISCUSSION TITLE
termux armv7 error install maturin

### DISCUSSION THREAD

**Discussion by Holmberg-15A** [PyO3/maturin#1639] [Q&A]
error install maturin:
error occurred: Failed to find tool. Is `arm-linux-androideabi-ar` installed?

            warning: build failed, waiting for other jobs to finish...
            error: `cargo build --manifest-path Cargo.toml --message-format=json-render-diagnostics --release -v --locked` failed with code 101
            [end of output]

        note: This error originates from a subprocess, and is likely not a problem with pip.
        ERROR: Failed building wheel for maturin
      Failed to build maturin
      ERROR: Could not build wheels for maturin, which is required to install pyproject.toml-based projects
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error
is it possible to install m mature?

**Comment by messense** [comment]
So... Is `arm-linux-androideabi-ar` installed?

**Reply by messense** [reply]
See https://github.com/termux/termux-packages/discussions/12913

```bash
pkg in binutils
```

**Comment by Holmberg-15A** [comment]
after installation binutils the error has changed: 
 💥 maturin failed
        Caused by: Cargo metadata failed. Does your crate compile with `cargo build`?
        Caused by: `cargo metadata` exited with an error:     Updating git repository `https://github.com/messense/cmake-rs.git`
      error: failed to load source for dependency `cmake`

      Caused by:
        Unable to update https://github.com/messense/cmake-rs.git?branch=cross-compile

      Caused by:
        failed to clone into: /data/data/com.termux/files/home/.cargo/git/db/cmake-rs-1a1c3c7fd02f9517

      Caused by:
        failed to authenticate when downloading repository

        * attempted to find username/password via git's `credential.helper` support, but failed

        if the git CLI succeeds then `net.git-fetch-with-cli` may help here
        https://doc.rust-lang.org/cargo/reference/config.html#netgit-fetch-with-cli

      Caused by:
        failed to acquire username/password from local configuration
      Error running maturin: Command '['maturin', 'pep517', 'write-dist-info', '--metadata-directory', '/data/data/com.termux/files/usr/tmp/pip-modern-metadata-8d72nido', '--interpreter', '/data/data/com.termux/files/usr/bin/python']' returned non-zero exit status 1.
      Checking for Rust toolchain....
      Running `maturin pep517 write-dist-info --metadata-directory /data/data/com.termux/files/usr/tmp/pip-modern-metadata-8d72nido --interpreter /data/data/com.termux/files/usr/bin/python`
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

**Reply by messense** [reply]
`https://github.com/messense/cmake-rs` repository was deleted long ago, please report the issue to the issue tracker of whatever package you were trying to install.

**Reply by messense** [reply]
BTW, please include full context when opening an issue in the future (in this or any other repositories) to save you and others time:

* What were you trying to achieve?
* What error did you encounter?
* Build environment information like OS version, software versions etc.

Thanks!

See also http://www.catb.org/~esr/faqs/smart-questions.html
